Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -1278,3 +1281,25 @@ extension ComposerVC: ChatChannelControllerDelegate {
         cooldownTracker.start(with: channelController.currentCooldownTime())
     }
 }
+
+private extension Array where Element == ChatMessageAttachment<Data> {
+    var toAnyAttachmentPayload: [AnyAttachmentPayload] {
+        let decoder = JSONDecoder()
+        return compactMap { attachment in
+            let type: AttachmentPayload.Type
+            switch attachment.type {
+            case .image: type = ImageAttachmentPayload.self
+            case .video: type = VideoAttachmentPayload.self
+            case .audio: type = AudioAttachmentPayload.self
+            case .file: type = FileAttachmentPayload.self
+            case .voiceRecording: type = VoiceRecordingAttachmentPayload.self
+            default:
+                log.assertionFailure("Unsupported attachment")
+                return nil
+            }
```

Comment:
Does this mean that customers can only edit attachments which are our own? Not custom attachments of the customers?

Reply:
An idea to overcome this is to having something like this:

```swift
           let types: [AttachmentType: AttachmentPayload.Type] = [
                .image: ImageAttachmentPayload.self,
                .video: VideoAttachmentPayload.self,
                .audio: AudioAttachmentPayload.self,
                .file: FileAttachmentPayload.self,
                .voiceRecording: VideoAttachmentPayload.self
            ]

            guard let payloadType = types[attachment.type] else { return nil }
            guard let payload = try? decoder.decode(payloadType, from: attachment.payload) else { return nil }
            return AnyAttachmentPayload(payload: payload)
```

The types would be a static property or something living in the `ChatClient` where customers would be able to register their own types. Something like this:
```ChatClient.shared.registerCustomAttachment(WorkoutAttachmentPayload.self)```

This is actually similar to what I was working a while ago when I was refactoring the Attachments API to a v2 because of some limitations when customers wanted to user their custom attachment and having integrated with our own download progress view here: https://github.com/GetStream/stream-chat-swift/pull/2459/files#diff-e77d975b26106706d818e66c3c924b6021b8f5e140ad008c99b394235791417eR40

Reply:
Moved to Slack: https://getstream.slack.com/archives/C03N8GWH8TZ/p1685975622681659

Reply:
Converted this to a follow-up task: https://github.com/GetStream/ios-issues-tracking/issues/463